File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max_ones=0,consecutive_ones=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==1){
                consecutive_ones++;
            }else{
                consecutive_ones=0;
            }
            max_ones=max(max_ones,consecutive_ones);
        }
        return max_ones;
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int currOnes=0,maxOnes=0;
        int n=nums.length;
        for(int i=0;i<n;i++){
            if(nums[i]==1){
                currOnes++;
            }else{
                currOnes=0;
            }
            maxOnes=Math.max(currOnes,maxOnes);
        }
        return maxOnes;
    }
}